---
title: "A detailed application of spaceMap"
output:
  html_document:
    highlight: tango
---

This webpage illustrates a detailed application of the [spaceMap network analysis pipeline](https://topherconley.github.io/spacemap/index.html) to real-world data from the [Breast Cancer Proteogenomics Landscape Study (BCPLS)](http://www.nature.com/nature/journal/v534/n7605/abs/nature18003.html). We show how to clean the data, fit spaceMap to learn networks, derive annotations, and ultimately layer biological meaning onto the model fit. Below, the flow chart provides an overview of each stage we will address in the pipeline: data, model fitting, network analysis, and visualization.

![](figures/flow-chart-4.png)

This specific application builds upon the original work of the BCPLS by performing an integrative analysis of copy number alterations (CNA) in breast tumors with corresponding protein expressions. The outcome of the integrative analysis is an annotated network revealing how CNA influence downstream protein-protein interactions of breast tumors. In addition to finding known amplifications and deletion signatures of CNA in breast cancer, the pipeline is able to link these CNA hubs to proteins enriched with cancer-related biological processes. All things considered, this pipeline exemplifies the power of reconstructing networks from data and leveraging bioinformatics toolkits to layer meaning onto the network. *Full commentary on the biological context of the BCPLS application is reserved exclusively for the spaceMap publication, which is currently under review*. 

<!--


## Data/Annotation



In order to understand the biological pathways that are perturbed by predictor nodes (e.g. CNA), the creation of a Gene  Ontology (or KEGG) universe is essential. By associating individual genes with their higher order biological processses, we can perform pathway analysis on the network. We illustrate how to derive such a universe in the [Gene Ontology tab](neta/prep-go-universes.html). 

## Model Fitting


##  Network Analysis

Deploying the network analysis toolkit is as simple as calling the `spacemap::neta` function. For now, there is one standard analysis that is implemented; however, the toolkit has an modular implementation, which makes it flexible to adapt the analysis to the future needs of users. The [Network Analysis tab](neta/neta-bcpls.html) illustrates the input and output of `spacemap::neta` with respect to the cleaned BCPLS data. 

## Visualization

-->

## R environment 

In addition to installing the [spacemap package](https://topherconley.github.io/spacemap/), reproduction of the BCPLS network analysis will require installation of the following R packages. 

```{r, eval = FALSE}
## try http:// if https:// URLs are not supported
source("https://bioconductor.org/biocLite.R")
BiocInstaller::biocLite()
biocLite(c("Biobase", "topGO", "GenomicRanges", "org.Hs.eg.db",
           "mygene"))
install.packages(c("ggplot2", "data.table", "fpc", 
                   "stringdist"))
```

