---
title: "An application of spaceMap to BCPLS data"
output:
  html_document:
    highlight: tango
---

This webpage illustrates a detailed application of the [spaceMap analysis pipeline](https://topherconley.github.io/spacemap/index.html) to the [Breast Cancer Proteogenomics Landscape Study (BCPLS)](http://www.nature.com/nature/journal/v534/n7605/abs/nature18003.html). We show how to [process data](https://topherconley.github.io/neta-bcpls/data.html), [infer networks](https://topherconley.github.io/neta-bcpls/mfits.html), [map annotations](https://topherconley.github.io/neta-bcpls/annot.html), and ultimately layer biological meaning onto the fitted networks through [network analysis](https://topherconley.github.io/neta-bcpls/neta.html). The flow chart below provides an overview of each step we will address in the pipeline: data, model fitting, network analysis, and visualization.

![](figures/flow-chart-4.png)


## Description 
This application  performs an integrative analysis of copy number alterations (CNA),  protein expressions, and RNA expressions in breast tumors. The outcomes are annotated networks revealing how CNAs influence downstream expression  interactions. This analysis links CNAs to protein/RNA expressions and finds CNA-hubs/modules that are enriched with cancer-related biological processes. In summary, this pipeline exemplifies the power of reconstructing networks from data and leveraging bioinformatics tools to layer meaning onto the network. 

*Full commentary on the biological context of the BCPLS application is reserved exclusively for the spaceMap publication, which is currently under review*. 

<!--


## Data/Annotation



In order to understand the biological pathways that are perturbed by predictor nodes (e.g. CNA), the creation of a Gene  Ontology (or KEGG) universe is essential. By associating individual genes with their higher order biological processses, we can perform pathway analysis on the network. We illustrate how to derive such a universe in the [Gene Ontology tab](neta/prep-go-universes.html). 

## Model Fitting


##  Network Analysis

Deploying the network analysis toolkit is as simple as calling the `spacemap::neta` function. For now, there is one standard analysis that is implemented; however, the toolkit has an modular implementation, which makes it flexible to adapt the analysis to the future needs of users. The [Network Analysis tab](neta/neta-bcpls.html) illustrates the input and output of `spacemap::neta` with respect to the cleaned BCPLS data. 

## Visualization

-->

## R environment 

In addition to installing the [spacemap package](https://topherconley.github.io/spacemap/), reproduction of the BCPLS data analysis will require installation of the following R packages: 

```{r, eval = FALSE}
## try http:// if https:// URLs are not supported
source("https://bioconductor.org/biocLite.R")
BiocInstaller::biocLite()
biocLite(c("Biobase", "topGO", "GenomicRanges", "org.Hs.eg.db",
           "mygene"))
install.packages(c("ggplot2", "data.table", "fpc", 
                   "stringdist"))
```

