<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Christopher Conley, Pei Wang, Umut Ozbek, Jie Peng" />

<meta name="date" content="2017-09-23" />

<title>BCPLS Marginal Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Application</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    BCPLS Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="data.html">1. Data</a>
    </li>
    <li>
      <a href="mfits.html">2. Model Fitting</a>
    </li>
    <li>
      <a href="annot.html">3. Annotation</a>
    </li>
    <li>
      <a href="neta.html">4. Network Analysis</a>
    </li>
    <li>
      <a href="marginal.html">5. Marginal Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/topherconley/neta-bcpls">
    <span class="fa fa-github"></span>
     
    BCPLS Code
  </a>
</li>
<li>
  <a href="https://topherconley.github.io/spacemap/">spacemap</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">BCPLS Marginal Analysis</h1>
<h4 class="author"><em>Christopher Conley, Pei Wang, Umut Ozbek, Jie Peng</em></h4>
<h4 class="date"><em>2017-09-23</em></h4>

</div>


<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>Marginal correlation analysis of proteogenomic breast cancer data is a computationally efficient method that carries the advantage of being applied genome-wide, while the graphical model spaceMap is not yet able to model genome-wide features. However, marginal analysis lacks the same modeling precision of graphical models and is prone to overfitting. The differences in feature resolution and number of features typically input to these two strategies make them difficult to compare. The following presentation compares the marginal analysis approach with spaceMap on a highly-variable subset of proteins and large copy number aberations intervals (CNA) of several megabases—expliclity using the same feature input. This comparison avoids comparing a marginal analysis with gene-level CNA input (as was done in the original BCPLS publication) with a spaceMap analysis with large genomic CNA. The outcome of this analysis clearly highlights the advantages spaceMap has over a marginal analysis when the feature input is the same.</p>
</div>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>Import the protein and the CNA data that was used to fit spaceMap to the breast cancer data associated with the BCPLS study.</p>
<pre class="r"><code>library(spacemap)
data(&quot;bcpls&quot;)
attach(bcpls)
#useful for reporting enriched hubs/modules later. 
library(AnnotationDbi)
library(GO.db)
process_alias &lt;- AnnotationDbi::Term(names(go2eg))
#read protein abundance and CNA data 
library(Biobase)
protset &lt;- readRDS(file = &quot;data/prot-expression-set.rds&quot;)
y &lt;- t(exprs(protset))
colnames(y) &lt;- yinfo$id
cnaset &lt;- readRDS(file = &quot;data/cna-expression-set.rds&quot;)
x &lt;- t(exprs(cnaset))
colnames(x) &lt;- xinfo$id</code></pre>
</div>
<div id="pearson-correlation-analysis" class="section level2">
<h2>Pearson Correlation Analysis</h2>
<p>We will compute the P-values from the Pearson correlation statistic for each CNA–protein and protein-protein pair with these function.</p>
<pre class="r"><code>xyPvalues &lt;- function(x, y, method = c(&quot;pearson&quot;, &quot;spearman&quot;), rdsfile) { 
  method = match.arg(arg = method, choices = c(&quot;pearson&quot;, &quot;spearman&quot;))
  require(foreach)
  pvals &lt;- foreach(p = seq_len(ncol(x)), .combine = &#39;rbind&#39;) %:% 
    foreach(q = seq_len(ncol(y)), .combine = &#39;c&#39;) %dopar% { 
      cor.test(x = x[,p], y = y[,q], method = method)$p.value
    }
  colnames(pvals) &lt;- colnames(y)
  rownames(pvals) &lt;- colnames(x)
  saveRDS(object = pvals, file = rdsfile)
  pvals
}

yyPvalues &lt;- function(y, method = c(&quot;pearson&quot;, &quot;spearman&quot;), rdsfile) { 
  method = match.arg(arg = method, choices = c(&quot;pearson&quot;, &quot;spearman&quot;))
  require(foreach)

  #generate lower triangle indices to iterate over
  rows &lt;- nrow(y)  
  z &lt;- sequence(rows)
  indices &lt;- cbind(  row = unlist(lapply(2:rows, function(x) x:rows), use.names = FALSE),
                     col = rep(z[-length(z)], times = rev(tail(z, -1))-1))
  
  pvals &lt;- foreach(j = seq_len(nrow(indices)), .combine = &#39;c&#39;) %dopar% {
    cor.test(x = y[,indices[j,&quot;row&quot;]], y = y[,indices[j,&quot;col&quot;]], method = method)$p.value
  }
  edge_list &lt;- data.frame(from = colnames(y)[indices[,&quot;row&quot;]], to = colnames(y)[indices[,&quot;col&quot;]], pvalue = pvals)
  saveRDS(object = edge_list, file = rdsfile)
  edge_list
}</code></pre>
<p>Compute the P-values for each protein-CNA pair and save them to file because this takes 20-30 minutes with 7 cores (pre-computed).</p>
<pre class="r"><code>pxyfile &lt;- &quot;~/scratch-data/neta-bcpls/marginal/xy_pearson_protein_bcpls.rds&quot;
stopifnot(!dir.exists(basename(pxyfile)))
library(doParallel)
cl &lt;- makeCluster(detectCores() - 1)
registerDoParallel(cl)
pxypval &lt;- xyPvalues(x = x, y = y, method = &quot;pearson&quot;, rdsfile = pxyfile)</code></pre>
<p>Compute the P-values for each protein–protein pair.</p>
<pre class="r"><code>pyyfile &lt;- &quot;~/scratch-data/neta-bcpls/marginal/yy_pearson_protein_bcpls.rds&quot;
stopifnot(!dir.exists(basename(pyyfile)))
pyypval &lt;- yyPvalues(y = y, method = &quot;pearson&quot;, rdsfile = pyyfile)</code></pre>
<p>Read in the pre-computed P-values from file.</p>
<pre class="r"><code>pxypval &lt;- readRDS(&quot;~/scratch-data/neta-bcpls/marginal/xy_pearson_protein_bcpls.rds&quot;)
pyypval &lt;- readRDS(&quot;~/scratch-data/neta-bcpls/marginal/yy_pearson_protein_bcpls.rds&quot;)</code></pre>
<p>Form the adjacency matrices of the marginal (Pearson correlation) network. Apply the Benjamini-Hochberg correction to every (CNA–protein, protein–protein) interaction and then keep only those interactions that pass an FDR &lt; 0.1. This method produces what we refer to hereafter as the “marginal network” from the significant protein-CNA interactions based on Pearson correlations (FDR &lt; 0.1). Note that an FDR threshold of 0.05 and 0.01 produce the same number of significant edges. An FDR of 0.1 is a reasonable threshold for exploratory network analyses such as this.</p>
<pre class="r"><code>adjMatCor &lt;- function(xp, dyp, x, y, thresh = 0.1) { 
  
  #partition and index x--y vs y--y
  xp &lt;- as.numeric(xp)
  yp &lt;- as.matrix(dyp$pvalue)
  thresh &lt;- 0.01
  nxp &lt;- length(xp)
  nyp &lt;- length(yp)
  xi &lt;- 1:nxp
  yi &lt;- (nxp + 1):(nxp +  nyp)
  
  #global adjustment
  gp &lt;- p.adjust(p = c(xp, yp), method = &quot;BH&quot;)
  gyp &lt;- gp[yi]
  gxp &lt;- gp[xi]
  #make x--y adjacency matrix
  mgxp &lt;- matrix(data = gxp, nrow = ncol(x), ncol = ncol(y))
  adjxy &lt;- (mgxp &lt;= thresh) + 0
  rownames(adjxy) &lt;- colnames(x)
  colnames(adjxy) &lt;- colnames(y)
  
  #make y--y adjacency matrix
  dyp$fdr &lt;- gyp
  make_yy_adj_matrix &lt;- function(el) { 
    ay &lt;- matrix(data = 0L, nrow = ncol(y), ncol = ncol(y))
    rownames(ay) &lt;- colnames(y)
    colnames(ay) &lt;- colnames(y)
    for(j in seq_len(nrow(el))) { 
      if (el[j,&quot;fdr&quot;] &lt;= thresh) { 
        ay[el[j,&quot;from&quot;], el[j,&quot;to&quot;]] &lt;- 1L
        ay[el[j,&quot;to&quot;], el[j,&quot;from&quot;]] &lt;- 1L
      }
    }
    ay
  }
  adjyy &lt;- make_yy_adj_matrix(el = dyp) 
  
  #return list of adjacency matrices
  list(yy = adjyy, xy = adjxy)
}
marg &lt;- adjMatCor(xp = pxypval, dyp = pyypval, x = x, y = y, thresh = 0.1)
#check to make feature labels are correct.
stopifnot(all(colnames(x) == xinfo$id), all(colnames(y) == yinfo$id))</code></pre>
</div>
<div id="compare-marginal-vs-spacemap-networks" class="section level2">
<h2>Compare marginal vs spaceMap networks</h2>
<p>Now we will compare the marginal network based on the Pearson correlation and spaceMap-based CNA–protein networks for breast cancer.</p>
<!--Since we will compare only the protein-CNA edges, we use the protein--protein edges from spaceMap's Boot.Vote network as a place holder for parameters required to convert the marginal network into igraph format with the function  `spaceMap::adj2igraph`. -->
<p>Form the marginal network.</p>
<pre class="r"><code>imarg &lt;- adj2igraph(yy = marg$yy, xy = marg$xy, yinfo = yinfo, xinfo = xinfo)</code></pre>
<p>Import spaceMap’s Boot.Vote network.</p>
<pre class="r"><code>pgraphml &lt;- &quot;/Users/bioinformatics/repos/neta-bcpls/neta/spacemap-prot-boot-vote.graphml&quot;
stopifnot(file.exists(pgraphml))
library(igraph)
ig &lt;- read_graph(file = pgraphml, format = &quot;graphml&quot;)</code></pre>
<p>Subset the marginal network and spaceMap Boot.Vote network into their <span class="math inline">\(y-y\)</span> and <span class="math inline">\(x-y\)</span> subnetworks to aid the comparisons. Intersect the two networks to see what is in common.</p>
<pre class="r"><code>imargxy &lt;- igraph::subgraph.edges(imarg, eids = E(imarg)[levels %in% &quot;x-y&quot;])
igxy &lt;- igraph::subgraph.edges(ig, eids = E(ig)[levels %in% &quot;x-y&quot;])
imargyy &lt;- igraph::subgraph.edges(imarg, eids = E(imarg)[levels %in% &quot;y-y&quot;])
igyy &lt;- igraph::subgraph.edges(ig, eids = E(ig)[levels %in% &quot;y-y&quot;])
# what is common
iixy &lt;- igraph::intersection(igxy, imargxy)
iiyy &lt;- igraph::intersection(igyy, imargyy)</code></pre>
<p>There are 669 CNA–protein edges in the marginal network compared to 585 in spaceMap’s network. The two networks have 59 CNA–protein edges in common. There are 416 protein–protein edges in the marginal network compared to 954 in spaceMap’s network. The two networks have 8 proteins–protein edges in common. Next we compare the CNA hub degree distributions between the two networks.</p>
<pre class="r"><code>#plot CNA hub degree distributions. 
library(ggplot2)
marg_cna_deg &lt;- degree(imargxy)[V(imargxy)[levels %in% &quot;x&quot;]]
ig_cna_deg &lt;- degree(igxy)[V(igxy)[levels %in% &quot;x&quot;]]
degcna &lt;- data.frame(degree = c(marg_cna_deg, ig_cna_deg),
                     network = rep(c(&quot;marginal&quot;, &quot;spaceMap&quot;), 
                                   times = c(length(marg_cna_deg),
                                             length(ig_cna_deg))))
ggplot(data = degcna, aes(x = degree, fill = network)) + 
  geom_histogram(binwidth = 3) + facet_grid(network ~ ., scale = &quot;free&quot;) + 
  xlab(&quot;CNA hub degree&quot;) + 
  theme_bw()</code></pre>
<p><img src="marginal_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>There are 234 CNA hubs in the marginal network with 61.1 percent having only one degree. There are 15 CNA hubs in the spaceMap network with 6.7 percent having only one degree. Next, we identify the specific CNA hubs for both networks.</p>
<pre class="r"><code>marg_xhubs &lt;- as_ids(V(imargxy)[igraph::degree(imargxy) &gt; 0 &amp; V(imargxy)$levels %in% &quot;x&quot;])
ig_xhubs &lt;- as_ids(V(igxy)[igraph::degree(igxy) &gt; 0 &amp; V(igxy)$levels %in% &quot;x&quot;])</code></pre>
<p>The 11 common CNA hubs between marginal and spaceMap networks are consistent with the original BCPLS publication, which reported “5q (loss of heterozygosity (LOH) in basal; gain in luminal B), 10p (gain in basal), 12 (gain in basal), 16q (luminal A deletion), 17q (luminal B amplification), and 22q (LOH in luminal and basal)”. Notably, only 22q is absent from the spaceMap analysis and only 3 small-degree CNA hubs (mean degree of 1.19) from 22q are represented by the marginal analysis. The evidence leaves open the possiblity that the the variance filter applied to the original data may have omitted proteins that are perturbed by 22q; or that the CNA hub-wise FDR filter applied in the original BCPLS study , rather than a global FDR correction, allowed the BCPLS authros to report more CNA hub regions.</p>
<table>
<thead>
<tr class="header">
<th align="left">Common CNA hub</th>
<th align="right">spaceMap degree</th>
<th align="right">marginal degree</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">5q34</td>
<td align="right">129</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">8q21.2-22.1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">10p15.1-15.3</td>
<td align="right">86</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">11q13.5-14.1</td>
<td align="right">11</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">12q21.1</td>
<td align="right">4</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">16p12.1-12.3</td>
<td align="right">21</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">16q22.1</td>
<td align="right">16</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">16q22.1-22.2</td>
<td align="right">72</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">17q12</td>
<td align="right">52</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">17q21.32</td>
<td align="right">99</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">17q23.1-23.2</td>
<td align="right">21</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<p>The CNA hubs unique to spaceMap (listed below) are associated with module P1, but are not associated with any specific GO-enriched proteins. Those CNA hubs are:</p>
<pre class="r"><code>#as_data_frame(x = ig, what = &quot;vertices&quot;)[setdiff(ig_xhubs, marg_xhubs),]
ucna_smap &lt;- xinfo$alias[match(setdiff(ig_xhubs, marg_xhubs), xinfo$id)]
ucna_smap</code></pre>
<pre><code>## [1] &quot;5p12-5q11.1&quot;  &quot;15q13.1-15.1&quot; &quot;17p11.2&quot;      &quot;17p11.2&quot;</code></pre>
<p>The CNA hubs unique to the marginal network have generally much smaller degree than the spaceMap network and are represented across 19 chromosomes. In the following table, we list chromosome with the frequency of reported CNA hubs <em>unique</em> to the marginal analysis. The mean degree of the CNA hubs in the marginal network is 2.8589744, while for spaceMap it is 39. The marginal network clearly reports more CNA perturbing the genome, but the effect is more diminished.</p>
<pre class="r"><code>ucna_marg &lt;- xinfo$alias[match(setdiff(marg_xhubs, ig_xhubs), xinfo$id)]
ucna_marg_df &lt;- data.frame(cytoband = ucna_marg, id = setdiff(marg_xhubs, ig_xhubs))
ucna_marg_tab &lt;- table(sapply(strsplit(ucna_marg, split = &quot;p|q&quot;), function(x) x[1]))
ucna_marg_tab2 &lt;- as.data.frame(ucna_marg_tab)
ucna_marg_tab2 &lt;- ucna_marg_tab2[with(ucna_marg_tab2, order(Freq, decreasing = T)),]
colnames(ucna_marg_tab2) &lt;- c(&quot;Chromosome&quot;, &quot;# of CNA hubs Unique to Marginal Analysis&quot;)
mean_chr_degree &lt;- function(chr) { mean(degree(imargxy)[as.character(ucna_marg_df$id[grep(chr, ucna_marg_df$cytoband)])]) }
ucna_marg_tab2$`Mean Degree` &lt;- round(sapply(ucna_marg_tab2$Chromosome, mean_chr_degree),2)
knitr::kable(ucna_marg_tab2, row.names = F)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Chromosome</th>
<th align="right"># of CNA hubs Unique to Marginal Analysis</th>
<th align="right">Mean Degree</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">17</td>
<td align="right">68</td>
<td align="right">1.76</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="right">34</td>
<td align="right">5.48</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="right">23</td>
<td align="right">2.27</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="right">15</td>
<td align="right">1.44</td>
</tr>
<tr class="odd">
<td align="left">16</td>
<td align="right">15</td>
<td align="right">1.50</td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="right">14</td>
<td align="right">2.90</td>
</tr>
<tr class="odd">
<td align="left">15</td>
<td align="right">12</td>
<td align="right">1.29</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="right">9</td>
<td align="right">1.00</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="right">8</td>
<td align="right">2.10</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="right">6</td>
<td align="right">1.58</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="right">5</td>
<td align="right">1.17</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="right">4</td>
<td align="right">2.00</td>
</tr>
<tr class="odd">
<td align="left">22</td>
<td align="right">3</td>
<td align="right">1.19</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="right">2</td>
<td align="right">2.12</td>
</tr>
<tr class="odd">
<td align="left">18</td>
<td align="right">1</td>
<td align="right">1.00</td>
</tr>
<tr class="even">
<td align="left">19</td>
<td align="right">1</td>
<td align="right">1.00</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="right">1</td>
<td align="right">4.04</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">1</td>
<td align="right">3.35</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="right">1</td>
<td align="right">1.75</td>
</tr>
</tbody>
</table>
</div>
<div id="marginal-hub-analysis" class="section level2">
<h2>Marginal Hub Analysis</h2>
<p>Rank the protein and CNA nodes, resptively.</p>
<pre class="r"><code>imarg &lt;- rankHub(ig = imarg, level = &quot;y&quot;)
imarg &lt;- rankHub(ig = imarg, level = &quot;x&quot;)</code></pre>
<p>Next label <span class="math inline">\(x-y\)</span> edges as being regulated in cis or in trans.</p>
<pre class="r"><code>library(GenomicRanges)
imarg &lt;- cisTrans(ig = imarg, level = &quot;x-y&quot;)</code></pre>
<p>We report the top 50 (degree-ranked) CNA hubs for the marginal network. The marginal analysis identifies much less than cis edges than one would expect, which is similar to the spaceMap network. Further the marginal network reports a large number of redundant CNA hubs compared to the spaceMap network, where redundant is defined to be CNA hubs of very close genomic distance and share very similar edges. For example there are at least 5 redundant 17q12 hubs, 2 redundant edges for 17q25, and 3 redundant edges for 6q21.</p>
<pre class="r"><code>xhubs &lt;- reportHubs(imarg, top = 50, level = &quot;x&quot;)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">hub</th>
<th align="left"># cis/ # trans</th>
<th align="right">Potential # cis</th>
<th align="left">cis genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">5q33.3-34 (160-160 Mb)</td>
<td align="left">0 / 23</td>
<td align="right">1</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.3 (180-180 Mb)</td>
<td align="left">0 / 22</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q34 (160-170 Mb)</td>
<td align="left">0 / 21</td>
<td align="right">0</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q14.3 (88-89 Mb)</td>
<td align="left">0 / 18</td>
<td align="right">2</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q34 (170-170 Mb)</td>
<td align="left">0 / 18</td>
<td align="right">0</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q33.1-33.3 (150-160 Mb)</td>
<td align="left">0 / 17</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q34-35.1 (170-170 Mb)</td>
<td align="left">0 / 17</td>
<td align="right">2</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.3 (180-180 Mb)</td>
<td align="left">0 / 17</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q35.2-35.3 (180-180 Mb)</td>
<td align="left">0 / 16</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q34 (170-170 Mb)</td>
<td align="left">0 / 15</td>
<td align="right">0</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q14.3 (89-91 Mb)</td>
<td align="left">0 / 14</td>
<td align="right">2</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.2 (170-180 Mb)</td>
<td align="left">0 / 14</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q11.2-14.3 (59-88 Mb)</td>
<td align="left">0 / 13</td>
<td align="right">15</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q23.2-23.3 (130-130 Mb)</td>
<td align="left">0 / 11</td>
<td align="right">4</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q14.3-23.2 (91-130 Mb)</td>
<td align="left">1 / 9</td>
<td align="right">12</td>
<td align="left">NUDT12</td>
</tr>
<tr class="even">
<td align="left">17q21.32 (46-46 Mb)</td>
<td align="left">0 / 10</td>
<td align="right">14</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q11.2 (58-59 Mb)</td>
<td align="left">0 / 9</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q23.2 (130-130 Mb)</td>
<td align="left">0 / 9</td>
<td align="right">3</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q23.3-33.1 (130-150 Mb)</td>
<td align="left">0 / 9</td>
<td align="right">16</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.2 (170-170 Mb)</td>
<td align="left">0 / 9</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17q21.32 (45-46 Mb)</td>
<td align="left">0 / 8</td>
<td align="right">15</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q11.2 (52-53 Mb)</td>
<td align="left">0 / 7</td>
<td align="right">3</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q11.2 (58-58 Mb)</td>
<td align="left">0 / 7</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q11.2 (53-56 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q11.2 (57-58 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.2 (170-170 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q35.2 (170-170 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">17q12 (38-38 Mb)</td>
<td align="left">4 / 2</td>
<td align="right">34</td>
<td align="left">ERBB2, GRB7, PNMT, MIEN1</td>
</tr>
<tr class="odd">
<td align="left">5q11.2 (56-57 Mb)</td>
<td align="left">0 / 5</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.1 (170-170 Mb)</td>
<td align="left">0 / 5</td>
<td align="right">2</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">10p15.1-15.3 (0.42-4 Mb)</td>
<td align="left">0 / 5</td>
<td align="right">10</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">16q22.1-22.2 (68-71 Mb)</td>
<td align="left">0 / 5</td>
<td align="right">13</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17q12 (37-37 Mb)</td>
<td align="left">3 / 2</td>
<td align="right">33</td>
<td align="left">ERBB2, GRB7, MIEN1</td>
</tr>
<tr class="even">
<td align="left">17q12 (37-38 Mb)</td>
<td align="left">3 / 2</td>
<td align="right">34</td>
<td align="left">ERBB2, GRB7, MIEN1</td>
</tr>
<tr class="odd">
<td align="left">17q12 (38-38 Mb)</td>
<td align="left">3 / 2</td>
<td align="right">34</td>
<td align="left">ERBB2, GRB7, MIEN1</td>
</tr>
<tr class="even">
<td align="left">5q35.1 (170-170 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">2</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q35.1-35.2 (170-170 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">4</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q35.2 (170-170 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17q12-21.1 (38-38 Mb)</td>
<td align="left">3 / 1</td>
<td align="right">34</td>
<td align="left">ERBB2, GRB7, MIEN1</td>
</tr>
<tr class="even">
<td align="left">17q21.1 (38-38 Mb)</td>
<td align="left">3 / 1</td>
<td align="right">34</td>
<td align="left">ERBB2, GRB7, MIEN1</td>
</tr>
<tr class="odd">
<td align="left">17q25.1 (72-73 Mb)</td>
<td align="left">4 / 0</td>
<td align="right">22</td>
<td align="left">SLC9A3R1, FDXR, FDXR, WBP2</td>
</tr>
<tr class="even">
<td align="left">17q25.1 (73-74 Mb)</td>
<td align="left">4 / 0</td>
<td align="right">22</td>
<td align="left">SLC9A3R1, FDXR, FDXR, WBP2</td>
</tr>
<tr class="odd">
<td align="left">17q25.1 (74-74 Mb)</td>
<td align="left">4 / 0</td>
<td align="right">22</td>
<td align="left">SLC9A3R1, FDXR, FDXR, WBP2</td>
</tr>
<tr class="even">
<td align="left">5q11.2 (57-57 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">6q16.3-21 (110-110 Mb)</td>
<td align="left">3 / 0</td>
<td align="right">4</td>
<td align="left">AIM1, PDSS2, C6orf203</td>
</tr>
<tr class="even">
<td align="left">6q21 (110-110 Mb)</td>
<td align="left">3 / 0</td>
<td align="right">4</td>
<td align="left">AIM1, PDSS2, C6orf203</td>
</tr>
<tr class="odd">
<td align="left">6q21 (110-110 Mb)</td>
<td align="left">3 / 0</td>
<td align="right">4</td>
<td align="left">AIM1, PDSS2, C6orf203</td>
</tr>
<tr class="even">
<td align="left">6q21 (110-110 Mb)</td>
<td align="left">3 / 0</td>
<td align="right">4</td>
<td align="left">AIM1, PDSS2, C6orf203</td>
</tr>
<tr class="odd">
<td align="left">6q23.1-23.2 (130-130 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">3</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">10p14-15.1 (4-12 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">13</td>
<td align="left">–</td>
</tr>
</tbody>
</table>
<p>The marginal network identified the 5q arm as having high degree and having also perturbed proteins with common functional roles according to the GO-neighbor percentage. This result conforms well to the spaceMap and original BCPLS analysis, which highlight the 5q arm loss in the basal molecular subtype.</p>
<pre class="r"><code>hgp &lt;- xHubEnrich(ig = imarg, go2eg = go2eg)
hgp2 &lt;- hgp[hgp$degree &gt; 10,]
kable(hgp2[order(hgp2$neighbor_percentage, decreasing = T),], row.names = FALSE)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">hub</th>
<th align="right">degree</th>
<th align="right">neighbor_percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">5q23.2-23.3</td>
<td align="right">11</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q33.1-33.3</td>
<td align="right">17</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">5q33.3-34</td>
<td align="right">23</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q34</td>
<td align="right">21</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">5q34</td>
<td align="right">18</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q34</td>
<td align="right">15</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">5q34-35.1</td>
<td align="right">17</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q35.2</td>
<td align="right">14</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">5q35.2-35.3</td>
<td align="right">16</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q35.3</td>
<td align="right">17</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">5q35.3</td>
<td align="right">22</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q11.2-14.3</td>
<td align="right">13</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">5q14.3</td>
<td align="right">18</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">5q14.3</td>
<td align="right">14</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="marginal-module-analysis" class="section level2">
<h2>Marginal Module Analysis</h2>
<p>Use the edge-betweenness algorithm to cluster.</p>
<pre class="r"><code>library(igraph)
mods &lt;- cluster_edge_betweenness(imarg)
mod_tab &lt;- table(mods$membership)</code></pre>
<p>There are 7 modules of size 15 or greater.</p>
<p>The marginal network GO-enrichment is very limited by any standard—finding only 2 GO-terms enriched—especially so when compared to spaceMap, which found over 40 GO-enriched terms. Both methods identify a common GO-enrichment of proteins involved in transcriptional regulation that are in trans-association with 5q.</p>
<pre class="r"><code>outmod &lt;- modEnrich(ig = imarg, mods = mods, levels = &quot;y&quot;, 
                    go2eg = go2eg, 
                    prefix = &quot;MARG&quot;, process_alias = process_alias)
knitr::kable(outmod$etab, row.names = FALSE)</code></pre>
<p>Export for visualization.</p>
<pre class="r"><code>filename &lt;- &quot;~/repos/neta-bcpls/neta/marginal-prot-boot-vote.graphml&quot;
#only keep vertices that have edges for visualization
vis &lt;- delete_vertices(graph = outmod$ig, v = V(outmod$ig)[igraph::degree(outmod$ig) == 0])
igraph::write_graph(graph = vis, file = filename, format = &quot;graphml&quot;)</code></pre>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>In comparing the marginal network to spaceMap’s Boot.Vote network, we see the advantages of spaceMap. The commonality of CNA hubs between the two analyses inspires confidence in the novel spaceMap method. Further, spaceMap reports less redundant and more prominent CNA hubs through the use of the group penalty term; this has the effect of potentially narrowing down the region of potential cancer drivers better than a marginal analysis and take this as evidence of spaceMap’s more direct modeling approach. In the <span class="math inline">\(y-y\)</span> edge space, there is much more GO-enrichment of the spaceMap method, showing it is more useful for extracting biological insight from high-dimensional proteogenomic data than marginal network analysis.</p>
</div>

<p>Copyright &copy; 2017 Regents of the University of California. All rights reserved.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
