<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Christopher Conley, Pei Wang, Jie Peng" />

<meta name="date" content="2017-09-21" />

<title>Network Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Application</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    BCPLS Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="data.html">1. Data</a>
    </li>
    <li>
      <a href="mfits.html">2. Model Fitting</a>
    </li>
    <li>
      <a href="annot.html">3. Annotation</a>
    </li>
    <li>
      <a href="neta.html">4. Network Analysis</a>
    </li>
    <li>
      <a href="marginal.html">5. Marginal Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/topherconley/neta-bcpls">
    <span class="fa fa-github"></span>
     
    BCPLS Code
  </a>
</li>
<li>
  <a href="https://topherconley.github.io/spacemap/">spacemap</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Network Analysis</h1>
<h4 class="author"><em>Christopher Conley, Pei Wang, Jie Peng</em></h4>
<h4 class="date"><em>2017-09-21</em></h4>

</div>


<p>Here we show how we applied the network analysis tolkit to <em>scggm-net</em>, a network that is described in Sections 4.2 and S.3.2 of the supplement to the spacemap publication. This document shares virtually identical analytical steps as the network analysis of <a href="https://topherconley.github.io/spacemap/articles/neta.html">spaceMap’s prot-net</a> and <a href="https://topherconley.github.io/neta-bcpls/neta.html">spaceMap’s RNA-net</a>.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Load the gene coordinate annotations for protein expressions and genomic CNA.</p>
<pre class="r"><code>library(Biobase)
yinfo &lt;- pData(featureData(readRDS(file = &quot;data/prot-expression-set.rds&quot;)))
xinfo &lt;- pData(featureData(readRDS(file = &quot;data/cna-expression-set.rds&quot;)))</code></pre>
<p>Load the <em>Boot.Vote</em> CNA-protein network.</p>
<pre class="r"><code>library(Matrix)
scggm &lt;- readRDS(file = &quot;model-fits/scggm_boot.rds&quot;)
net &lt;- scggm[c(&quot;yy&quot;,&quot;xy&quot;)]
rownames(net$xy) &lt;- xinfo$id; colnames(net$xy) &lt;- yinfo$id;
rownames(net$yy) &lt;- yinfo$id; colnames(net$yy) &lt;- yinfo$id;</code></pre>
<p>Load the degree distributions for the ensemble of bootstrapped networks.</p>
<pre class="r"><code>bdeg &lt;- scggm[c(&quot;deg_xy&quot;, &quot;deg_yy&quot;)]
names(bdeg) &lt;- c(&quot;xy&quot;, &quot;yy&quot;)
colnames(bdeg$xy) &lt;- xinfo$id; colnames(bdeg$yy) &lt;- yinfo$id;</code></pre>
<p>Load the Gene Ontology mappings for enrichment analysis.</p>
<pre class="r"><code>go2eg &lt;- readRDS(&quot;data/prot-go-bp-to-entrez-gene-list.rds&quot;)
library(AnnotationDbi)
#human readable 
process_alias &lt;- Term(names(go2eg))</code></pre>
</div>
<div id="map-annotations" class="section level2">
<h2>Map Annotations</h2>
<p>Convert the Boot.Vote network into an <em>igraph</em> object and map the annotations onto the network.</p>
<pre class="r"><code>library(spacemap)
ig &lt;- spacemap::adj2igraph(yy = net$yy, xy = net$xy, yinfo = yinfo, xinfo = xinfo)</code></pre>
<p>If we query the attribute names of the nodes in the graph, we notice that the columns of <code>xinfo</code> and <code>yinfo</code> have been applied.</p>
<pre class="r"><code>vertex_attr_names(graph = ig)</code></pre>
<pre><code>## [1] &quot;name&quot;        &quot;alias&quot;       &quot;chr&quot;         &quot;start&quot;       &quot;end&quot;        
## [6] &quot;strand&quot;      &quot;description&quot; &quot;levels&quot;</code></pre>
<p>The igraph package has a number of ways to access the annotation information. For example, if we wish to confirm the chromosome location of GRB7, we can easily query:</p>
<pre class="r"><code>vertex_attr(graph = ig, name = &quot;chr&quot;, index = V(ig)[alias %in% &quot;GRB7&quot;])</code></pre>
<pre><code>## [1] &quot;chr17&quot;</code></pre>
</div>
<div id="hub-analysis" class="section level2">
<h2>Hub Analysis</h2>
<p>We first prioritize the CNA- and protein- hubs. If the <code>bdeg</code> argument is specified, then we rank the hubs according to the average degree rank. Accordingly, the most highly ranked hubs will have the most consistently high degree across network ensemble.</p>
<p>To rank the protein nodes, use the <code>rankHub</code> command and simply specify the <code>level = &quot;y&quot;</code> argument.</p>
<pre class="r"><code>ig &lt;- rankHub(ig = ig, bdeg = bdeg$yy, level = &quot;y&quot;)</code></pre>
<p>To rank the CNA nodes, specify the <code>level = &quot;x&quot;</code> argument.</p>
<pre class="r"><code>ig &lt;- rankHub(ig = ig, bdeg = bdeg$xy, level = &quot;x&quot;)</code></pre>
<p>Alternatively, if the <code>bdeg</code> argument is not available, the ranking is according to degree in the final network (coded by “ig”).</p>
<pre class="r"><code>tmp &lt;- rankHub(ig = ig,  level = &quot;x&quot;)</code></pre>
<div id="identify-cis-and-trans" class="section level3">
<h3>Identify cis and trans</h3>
<p>Next label <span class="math inline">\(x-y\)</span> edges as being regulated in cis or in trans. The <a href="http://bioconductor.org/packages/release/bioc/html/GenomicRanges.html">GenomicRanges</a> R package and the genomic coordinates <code>chr,start,end</code> columns of <code>xinfo</code> and <code>yinfo</code> are required for this step.</p>
<p>Now we can label the <span class="math inline">\(x-y\)</span> edges with either “cis” or “trans” in the <code>cis_trans</code> edge attribute of <code>ig</code>.</p>
<pre class="r"><code>ig &lt;- cisTrans(ig = ig, level = &quot;x-y&quot;, cw = 2e6)</code></pre>
</div>
<div id="report" class="section level3">
<h3>Report</h3>
<p>We then report a well-organized table, as seen in Table 3 of the spaceMap publication. The <code>top</code> argument allows us to control how many hubs are displayed.</p>
<pre class="r"><code>xhubs &lt;- reportHubs(ig, top = 40, level = &quot;x&quot;)</code></pre>
<pre class="r"><code>vidx &lt;- V(ig)[match(rownames(xhubs), V(ig)$name)]
m &lt;- vertex_attr(graph = ig, name = &quot;mean_rank_hub&quot;, index = vidx)
s &lt;- vertex_attr(graph = ig, name = &quot;sd_rank_hub&quot;, index = vidx)
st &lt;- vertex_attr(graph = ig, name = &quot;start&quot;, index = vidx)
end &lt;- vertex_attr(graph = ig, name = &quot;end&quot;, index = vidx)
rnk &lt;- paste0(round(m, 2), &quot; (&quot;, round(s,2), &quot;)&quot;)
xhubs$`cis genes`
tosupp &lt;- cbind(xhubs$hub, st, end, xhubs$`# cis/ # trans`, xhubs$`Potential # cis`, xhubs$`cis genes`, rnk)
library(xtable)
print(xtable(tosupp), include.rownames = FALSE)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">hub</th>
<th align="left"># cis/ # trans</th>
<th align="right">Potential # cis</th>
<th align="left">cis genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">16q22.1-22.2 (68-71 Mb)</td>
<td align="left">0 / 17</td>
<td align="right">13</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">17q12 (38-38 Mb)</td>
<td align="left">4 / 6</td>
<td align="right">29</td>
<td align="left">ERBB2, GRB7, PNMT, MIEN1</td>
</tr>
<tr class="odd">
<td align="left">16q21-22.1 (64-68 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">10p15.1-15.3 (0.42-4 Mb)</td>
<td align="left">0 / 12</td>
<td align="right">10</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5p12-5q11.1 (45-50 Mb)</td>
<td align="left">0 / 13</td>
<td align="right">4</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">16q23.1-24.3 (78-89 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">11</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17p11.2 (20-20 Mb)</td>
<td align="left">1 / 5</td>
<td align="right">16</td>
<td align="left">SHMT1</td>
</tr>
<tr class="even">
<td align="left">5q35.3 (180-180 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">16p12.1-12.3 (19-27 Mb)</td>
<td align="left">1 / 5</td>
<td align="right">19</td>
<td align="left">DCTPP1</td>
</tr>
<tr class="even">
<td align="left">17p11.2 (20-21 Mb)</td>
<td align="left">0 / 10</td>
<td align="right">16</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">16q12.1 (51-53 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">6</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">15q13.1-15.1 (29-43 Mb)</td>
<td align="left">1 / 1</td>
<td align="right">20</td>
<td align="left">IVD</td>
</tr>
<tr class="odd">
<td align="left">1q21.2-23.1 (150-160 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">47</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">17p11.2-12 (14-19 Mb)</td>
<td align="left">0 / 5</td>
<td align="right">17</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">16q22.1 (68-68 Mb)</td>
<td align="left">0 / 8</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">19p13.3 (1.9-2.4 Mb)</td>
<td align="left">0 / 2</td>
<td align="right">16</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">16q22.2-22.3 (72-73 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">10</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">16p11.2-12.1 (27-32 Mb)</td>
<td align="left">1 / 0</td>
<td align="right">18</td>
<td align="left">DCTPP1</td>
</tr>
<tr class="odd">
<td align="left">8q21.2-22.1 (87-96 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">13</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q14.3 (88-89 Mb)</td>
<td align="left">0 / 6</td>
<td align="right">2</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q35.2-35.3 (180-180 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">5q34 (160-170 Mb)</td>
<td align="left">0 / 21</td>
<td align="right">0</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">16q21 (63-64 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">4</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">16q12.1 (49-51 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">5</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17q11.2 (28-29 Mb)</td>
<td align="left">3 / 0</td>
<td align="right">14</td>
<td align="left">KRT23, BLMH, NSRP1</td>
</tr>
<tr class="even">
<td align="left">8q24.3 (140-150 Mb)</td>
<td align="left">1 / 2</td>
<td align="right">7</td>
<td align="left">THEM6</td>
</tr>
<tr class="odd">
<td align="left">8p11.23 (38-38 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">9</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">11q22.2-22.3 (100-100 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">4</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17p11.2 (19-19 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">16</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">17q21.32 (46-46 Mb)</td>
<td align="left">2 / 2</td>
<td align="right">15</td>
<td align="left">MAPT, MAPT</td>
</tr>
<tr class="odd">
<td align="left">17p11.2 (19-20 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">16</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">11q22.1 (100-100 Mb)</td>
<td align="left">0 / 2</td>
<td align="right">3</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">9p13.3 (33-34 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">9</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">11q14.3 (89-91 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">5</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">17q12 (38-38 Mb)</td>
<td align="left">2 / 3</td>
<td align="right">29</td>
<td align="left">GRB7, MIEN1</td>
</tr>
<tr class="even">
<td align="left">22q13.1-13.2 (40-41 Mb)</td>
<td align="left">0 / 2</td>
<td align="right">7</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q35.2 (170-180 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">8</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">9p13.3 (34-36 Mb)</td>
<td align="left">0 / 4</td>
<td align="right">9</td>
<td align="left">–</td>
</tr>
<tr class="odd">
<td align="left">5q34 (170-170 Mb)</td>
<td align="left">0 / 1</td>
<td align="right">0</td>
<td align="left">–</td>
</tr>
<tr class="even">
<td align="left">11p15.2 (13-14 Mb)</td>
<td align="left">0 / 3</td>
<td align="right">1</td>
<td align="left">–</td>
</tr>
</tbody>
</table>
<p>Similarly, we can report the top 10 protein hubs, their degrees in the final network, and a description of each hub, if the <code>description</code> column was specified in <code>yinfo</code>.</p>
<pre class="r"><code>yhubs &lt;- reportHubs(ig, top = 10, level = &quot;y&quot;)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">hub</th>
<th align="right">degree</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PLIN1</td>
<td align="right">13</td>
<td align="left">perilipin-1</td>
</tr>
<tr class="even">
<td align="left">ADH1B</td>
<td align="right">10</td>
<td align="left">alcohol dehydrogenase 1B</td>
</tr>
<tr class="odd">
<td align="left">PDLIM7</td>
<td align="right">8</td>
<td align="left">PDZ and LIM domain protein 7 isoform 4</td>
</tr>
<tr class="even">
<td align="left">GPD1</td>
<td align="right">13</td>
<td align="left">glycerol-3-phosphate dehydrogenase</td>
</tr>
<tr class="odd">
<td align="left">PDLIM7</td>
<td align="right">11</td>
<td align="left">PDZ and LIM domain protein 7 isoform 1</td>
</tr>
<tr class="even">
<td align="left">EHD2</td>
<td align="right">23</td>
<td align="left">EH domain-containing protein 2</td>
</tr>
<tr class="odd">
<td align="left">KNG1</td>
<td align="right">28</td>
<td align="left">kininogen-1 isoform 1 precursor</td>
</tr>
<tr class="even">
<td align="left">KNG1</td>
<td align="right">26</td>
<td align="left">kininogen-1 isoform 2 precursor</td>
</tr>
<tr class="odd">
<td align="left">ITIH4</td>
<td align="right">19</td>
<td align="left">inter-alpha-trypsin inhibitor heavy chain H4 isoform 2 precursor</td>
</tr>
<tr class="even">
<td align="left">ITIH4</td>
<td align="right">17</td>
<td align="left">inter-alpha-trypsin inhibitor heavy chain H4 isoform 1 precursor</td>
</tr>
</tbody>
</table>
</div>
<div id="go-neighbor-percentage" class="section level3">
<h3>GO-neighbor percentage</h3>
<p>A <em>CNA neighborhood</em> comprises all protein nodes that are directly connected to a CNA hub by an edge. CNA neighborhoods represent direct perturbations to the proteome by amplifications or deletions in the DNA. To quantify their functional relevance, we compute a score called the <em>GO-neighbor percentage</em>. Two protein nodes are called GO-neighbors if they share a common GO term in the same CNA neighborhood. We postulate that a high percentage of GO-neighbors within a CNA neighborhood associates the CNA hub with more functional meaning. These scores can be generated with a GO mapping to the proteins as follows.</p>
<pre class="r"><code>hgp &lt;- xHubEnrich(ig = ig, go2eg = go2eg)</code></pre>
</div>
</div>
<div id="module-analysis" class="section level2">
<h2>Module Analysis</h2>
<p>There are many criteria to define modules of a network. This toolkit allows users to import the module membership information by themselves (see <code>mods</code> argument in <a href="https://topherconley.github.io/spacemap/reference/modEnrich.html">modEnrich</a>).</p>
<p>In the spaceMap publication, we use the edge-betweenness algorithm in <em>igraph</em>.</p>
<pre class="r"><code>library(igraph)
mods &lt;- cluster_edge_betweenness(ig)</code></pre>
<p>The main goal of module analysis is identifying modules that are functionally enriched. The <code>modEnrich</code> function will test for significantly over-represented GO-terms (or any other valid functional grouping) within a module using hyper-geometric tests.</p>
<p>In the current example, only the protein nodes have functional mapping and we specify this through the <code>levels = &quot;y&quot;</code> argument. If both predictors and responses have functional mapping in the <code>go2eg</code> argument, then we can specify <code>levels = c(&quot;y&quot;,&quot;x&quot;)</code>. Other arguments are available to control the enrichment testing; see <a href="https://topherconley.github.io/spacemap/reference/modEnrich.html">modEnrich</a> for more details.</p>
<pre class="r"><code>outmod &lt;- modEnrich(ig = ig, mods = mods, levels = &quot;y&quot;, go2eg = go2eg, process_alias = process_alias, prefix = &quot;C&quot;)</code></pre>
<p>The output of the module analysis is a list of 3 objects.</p>
<pre class="r"><code>names(outmod)</code></pre>
<pre><code>## [1] &quot;ig&quot;   &quot;etab&quot; &quot;eraw&quot;</code></pre>
<ul>
<li><p><code>ig</code> is the input igraph network object updated with a “process_id” attribute for nodes affiliated with a significant GO-term. The “process_id” and “module” attributes together are useful for visualizing which nodes are enriched for a specific biological function.</p></li>
<li><p><code>etab</code> is the polished module enrichment table to report significant GO terms, the representation of the GO term in the module relative to the size of the GO term, and which CNA hubs belong to the module. The top ten hits as in Table S.9 of the spaceMap publication’s supplementary materials are as follows:</p></li>
</ul>
<pre class="r"><code>outmod$etab[1:20,]</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Module (size)</th>
<th align="left">GO Category</th>
<th align="left">GO Obs./ Total</th>
<th align="left">X-hubs (hits)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C10 (21)</td>
<td align="left">extracellular matrix organization</td>
<td align="left">11/125</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">glycosaminoglycan metabolic process</td>
<td align="left">5/15</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">collagen catabolic process</td>
<td align="left">6/29</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">extracellular matrix disassembly</td>
<td align="left">7/57</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">cell adhesion</td>
<td align="left">8/105</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">carbohydrate metabolic process</td>
<td align="left">5/43</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">axon guidance</td>
<td align="left">5/58</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">cytokine-mediated signaling pathway</td>
<td align="left">5/59</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">C4 (17)</td>
<td align="left">ER to Golgi vesicle-mediated transport</td>
<td align="left">5/20</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">protein N-linked glycosylation via asparagine</td>
<td align="left">5/20</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">membrane organization</td>
<td align="left">5/23</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">post-translational protein modification</td>
<td align="left">5/43</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">axon guidance</td>
<td align="left">5/58</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">cellular protein metabolic process</td>
<td align="left">6/99</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">C5 (34)</td>
<td align="left">muscle organ development</td>
<td align="left">5/17</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">cell-matrix adhesion</td>
<td align="left">5/35</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">C6 (68)</td>
<td align="left">transcription from RNA polymerase II promoter</td>
<td align="left">8/20</td>
<td align="left">5q34 (20); 10p15.1-15.3 (12)</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">positive regulation of transcription from RNA polymerase II promoter</td>
<td align="left">9/52</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">C7 (47)</td>
<td align="left">muscle contraction</td>
<td align="left">13/26</td>
<td align="left">6q25.3 (5)</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">movement of cell or subcellular component</td>
<td align="left">7/15</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<ul>
<li><code>eraw</code> contains details for each (module, GO-term) pair that was subjected to the hyper-geometric test. This output gives the user more control by reporting all tests, the relative over-representation of a GO-term in that module, the raw P-value, and the adjusted P-value.</li>
</ul>
<pre class="r"><code>outmod$eraw[1:5,]</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Module</th>
<th align="left">process_id</th>
<th align="left">process_alias</th>
<th align="left">GO Obs./Total</th>
<th align="right">P_value</th>
<th align="right">Adjusted_P_value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C1</td>
<td align="left"><a href="GO:0007596" class="uri">GO:0007596</a></td>
<td align="left">blood coagulation</td>
<td align="left">5/105</td>
<td align="right">0.0019854</td>
<td align="right">0.0675412</td>
</tr>
<tr class="even">
<td align="left">C1</td>
<td align="left"><a href="GO:0002576" class="uri">GO:0002576</a></td>
<td align="left">platelet degranulation</td>
<td align="left">3/35</td>
<td align="right">0.0036786</td>
<td align="right">0.1089927</td>
</tr>
<tr class="odd">
<td align="left">C1</td>
<td align="left"><a href="GO:0001889" class="uri">GO:0001889</a></td>
<td align="left">liver development</td>
<td align="left">2/16</td>
<td align="right">0.0091840</td>
<td align="right">0.2219862</td>
</tr>
<tr class="even">
<td align="left">C1</td>
<td align="left"><a href="GO:0030168" class="uri">GO:0030168</a></td>
<td align="left">platelet activation</td>
<td align="left">3/51</td>
<td align="right">0.0106856</td>
<td align="right">0.2467104</td>
</tr>
<tr class="odd">
<td align="left">C1</td>
<td align="left"><a href="GO:0051260" class="uri">GO:0051260</a></td>
<td align="left">protein homooligomerization</td>
<td align="left">2/20</td>
<td align="right">0.0142285</td>
<td align="right">0.2774912</td>
</tr>
</tbody>
</table>
</div>
<div id="export-for-visualization" class="section level2">
<h2>Export for Visualization</h2>
<p>There are many tools for network visualization. In the publication of spaceMap, we exported the annotated <em>igraph</em> network object <code>ig</code> to the “graphml” format, which maintained all the attributes associated with nodes when read into <em>Cytoscape</em> for visualization.</p>
<pre class="r"><code>filename &lt;- &quot;~/repos/neta-bcpls/neta/spacemap-scggm-boot-vote.graphml&quot;
#delete nodes without edges from the graph object
vis &lt;- delete_vertices(graph = outmod$ig, v = V(outmod$ig)[igraph::degree(outmod$ig) == 0])
igraph::write_graph(graph = vis, file = filename, format = &quot;graphml&quot;)</code></pre>
<p>Here we list all the attributes associated with the nodes that can be used in tandem with Cytoscape’s filtering functions to identify specific nodes of interest.</p>
<pre class="r"><code>vertex_attr_names(outmod$ig)</code></pre>
<pre><code>##  [1] &quot;name&quot;             &quot;alias&quot;            &quot;chr&quot;             
##  [4] &quot;start&quot;            &quot;end&quot;              &quot;strand&quot;          
##  [7] &quot;description&quot;      &quot;levels&quot;           &quot;rank_hub&quot;        
## [10] &quot;mean_rank_hub&quot;    &quot;sd_rank_hub&quot;      &quot;ncis&quot;            
## [13] &quot;ntrans&quot;           &quot;regulates_in_cis&quot; &quot;potential_cis&quot;   
## [16] &quot;module&quot;           &quot;C&quot;</code></pre>
<p>We describe some of the most useful attributes for visualization:</p>
<ul>
<li>‘name’: the unique node ID</li>
<li>‘alias’: the node alias (e.g. gene symbol ERBB2)</li>
<li>‘chr,start,end,strand’: the gene coordinates of nodes</li>
<li>‘description’: any note (e.g. breast cancer oncogene)</li>
<li>‘levels’: indicates whether the node belongs to predictors “x” or responses “y”</li>
<li>‘rank_hub’: the rank of the hub within its level (e.g. a value of “1” for a node of level “x” means that it is the most consistently high degree <span class="math inline">\(x\)</span> node in the network. )</li>
<li>‘regulates_in_cis’: list of genes regulated in cis</li>
<li>‘module’: the module ID that the node belongs to.</li>
<li>‘process_id’: the significant GO-term(s) associated with the node.</li>
</ul>
<p>Also the edge attributes are exported to ‘graphml’ format.</p>
<pre class="r"><code>edge_attr_names(outmod$ig)</code></pre>
<pre><code>## [1] &quot;levels&quot;    &quot;cis_trans&quot;</code></pre>
<ul>
<li>‘levels’ indicates whether an edge is <span class="math inline">\(x-y\)</span> or <span class="math inline">\(y-y\)</span>.</li>
<li>‘cis_trans’ indicates whether an edge is regulated in cis or in trans.</li>
</ul>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>This toolkit is useful for analyzing and summarizing the output of the spaceMap model fit in the context of integrative genomics. The biological context mapped onto the network object can easily be exported to existing network visualization tools like <em>Cytoscape</em>.</p>
</div>

<p>Copyright &copy; 2017 Regents of the University of California. All rights reserved.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
